src/sentry/snuba/metrics/extraction.py

```diff
@@ -900,6 +900,8 @@ def _remove_redundant_parentheses(tokens: Sequence[QueryToken]) -> Sequence[Quer
 def _deep_sorted(value: Union[Any, Dict[Any, Any]]) -> Union[Any, Dict[Any, Any]]:
     if isinstance(value, dict):
         return {key: _deep_sorted(value) for key, value in sorted(value.items())}
+    elif isinstance(value, list):
+        return sorted(value, key=lambda x: (x.get("key", "")))
```

Comment:
Is `key` always present in the arrays we are sorting? This code assumes that every `list` is a `list[dict]` which might cause errors.